### ISSUE TITLE
Global Styles not using some selected fonts in certain themes

### ISSUE THREAD

**Issue by ClassicRKR27** [Automattic/wp-calypso#71641]
### Quick summary

A user is trying to update their Headings font on Zoologist to `Montserrat`, but the changes are not showing on the live site even though they are saving correctly in the Site Editor.

I can replicate this on my Simple Site as shown here: 

https://user-images.githubusercontent.com/51870544/210429672-9f3000db-f652-477b-b0b4-1b8e74e769e5.mp4

I tried to check on an atomic site, but it's only showing two fonts for some reason (possibly related to GDPR? not sure):
![SQ8OxV.png](https://user-images.githubusercontent.com/51870544/210429605-7304762e-a921-4c08-8cd8-5ca8f605f6b4.png)

### Steps to reproduce

1. Set a simple site to `Zoologist` theme
2. Try to change the font in the `Global Styles` to anything other than default
3. Check to see if the font updated on the live site

**Important: ensure that no individual blocks on the page are using the same font as you selected in global styles**. If an individual block on the page is using the same font as you selected in global styles then the font is loaded and it works.

### What you expected to happen

The font should show correctly after saving the Global Styles changes

### What actually happened

The changes are saved, but not reflected on the live site - they only show as default serif/sans-serif fonts

### Context

36494970-hc

Follow-up ticket to different address: 5832285-zd-woothemes

### Platform (Simple, Atomic, or both?)

Simple

### Theme-specific issue?

Any block theme when using a Jetpack supplied font. e.g. TT3

Any block-based derived theme, specifically Zoologist

### Browser, operating system and other notes

Firefox, Safari, Chrome

### Reproducibility

Consistent

### Severity

Most (> 50%)

### Available workarounds?

No but the platform is still usable

### Workaround details

If the page uses the font on a specific block rather than via global styles then the font will be loaded. Consider changing a heading or paragraph block in the header or footer to use the same font.

**Comment by dsas** [comment]
I can reproduce this. The CSS stored in the global styles is being output and the heading elements are have a computed font-family of "Montserrat, sans-serif" according to the web inspector. There's no network requests for the Montserrat font though. Font requests are handled via jetpack, the only thing I can see related in the page source is:

```
<style id='jetpack-global-styles-frontend-style-inline-css'>
@import url('https://fonts-api.wp.com/css?family=Fira Sans:thin,extralight,light,regular,medium,semibold,bold,italic,bolditalic,extrabold,black|Playfair Display:thin,extralight,light,regular,medium,semibold,bold,italic,bolditalic,extrabold,black|');:root { --font-headings: Playfair Display; --font-base: Fira Sans; --font-headings-default: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif; --font-base-default: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;}
</style>
```

**Comment by dsas** [comment]
_Probably_ [this Core issue](https://github.com/WordPress/gutenberg/issues/43397), but given customer reports only started in December afaict, maybe we've changed something since then to exacerbate it?

**Comment by dsas** [comment]
_Some_ of the reports on https://github.com/Automattic/wp-calypso/issues/67641 are actually this issue

@pbking has also pointed out that: 

> “Headings” hasn’t always been an element option in the Site Editor and probably became more obvious when that was added as it’s the most obviously effected.

**Comment by metabreakr** [comment]
I can also reproduce this with Twenty Twenty Three. I created another demo site yesterday, and selecting anything other than "Default" or "System" displays a serif font.

**Comment by Copons** [comment]
Just a quick reminder that @dsas added a workaround to the OP:

> ### Workaround details
> If the page uses the font on a specific block rather than via global styles then the font will be loaded. Consider changing a heading or paragraph block in the header or footer to use the same font.

It was originally mentioned by @pbking in https://github.com/WordPress/gutenberg/issues/43397, and it technically works because:

- GS sets the correct font but stores its name instead of the slug. The font name does not match the filename, and the enqueue fails.
- Regular blocks (correctly) store the font slug, which matches the font filename: the enqueue works.

Basically, the regular block would act as a "fallback enqueue" if the GS system fails.

**Comment by danielbachhuber** [comment]
Possibly related: pdKhl6-16Z-p2#comment-1260

**Comment by hanananah** [comment]
Another possible case: 26812607-hc

**Comment by ersatzpenguin** [comment]
Another possible instance: 5838628-zd-woothemes

**Comment by carinapilar** [comment]
This doesn't seem to be specific to a theme, I was able to replicate using other themes. Got a user on 37441174-hc with this issue using Rainfall. User would like an update once this is fixed.

**Comment by dsas** [comment]
@carinapilar It's not specific to a theme, which fonts will work depend upon the theme. I've updated the title to be less specific.

**Comment by arinoch** [comment]
Found an atomic site running Munchies that appears to be having this issue. The user was going for josefin sans lite, and getting josefin sans thin or Arial, depending on the thing it's applied to. Applying the workaround to individual blocks on this page does not solve the problem (the user tried that before contacting us).

39360544-hc

**Comment by happychait** [comment]
To be followed up in 5838577-zen as well.

**Comment by dsas** [comment]
> Found an atomic site running Munchies that appears to be having this issue. The user was going for josefin sans lite, and getting josefin sans thin or Arial, depending on the thing it's applied to. Applying the workaround to individual blocks on this page does not solve the problem (the user tried that before contacting us).

Some of that doesn't add up @arinoch. Looking at the CSS it's not even trying to use Josefin sans (which is being loaded) on the two buttons mentioned, it's trying and failing to use Inter: 
![image](https://user-images.githubusercontent.com/93301/210975091-94c64280-9201-4418-a453-f1c88fe4faba.png). While this issue is likely causing Inter not to load, I'd be less certain they have applied the workaround correctly, there might be some other misunderstanding or bug to iron out.

**Comment by rinazrina** [comment]
Another report on #5842151-zen
Livro theme.

**Comment by zdenys** [comment]
5838695-zd-woothemes Livro theme

**Comment by RealCarFax** [comment]
Another report: 39402024-hc

**Comment by happychait** [comment]
Another report in 39368086-hc

**Comment by Robertght** [comment]
5843053-zen 

The site is using Munchies as its theme.

**Comment by liviopv** [comment]
- [ ] 5843102-zd-woothemes

**Comment by Robertght** [comment]
5845668-zen

**Comment by jartes** [comment]
5838136-zd-woothemes

**Comment by thekingsprojects** [comment]
Another here: 5845512-zen

In this case the user wants to use both Work Sans and Space Mono, but Space Mono displays as default serif (Work Sans is fine and is what they're using on the whole site for now).

**Comment by rw-ye** [comment]
5847193-zd-woothemes

Theme: Masu

**Comment by geekinthegirl** [comment]
Another one here that may be related: 5846142-zd-woothemes

User's trying to use the Montserrat font, and currently has the Twenty Twenty Three theme applied.

**Comment by nerdysandy** [comment]
Reported on 5838238 -zen

**Comment by dolgelukkig** [comment]
5848068-zen. looks like it's related.

**Comment by louinel** [comment]
5840854-zen

**Comment by Copons** [comment]
Hi HEs!
A heads up that we have a PR to fix this issue open in Core Gutenberg: https://github.com/WordPress/gutenberg/pull/46944

We don't have an ETA, but we'll try to expedite the fix as quickly as we possibly can.

**Comment by upwardmomentum84** [comment]
Reported here: 37755131-hc

**Comment by carolframen** [comment]
#5848406-zen

**Comment by cintacks** [comment]
Another example while waiting on the PR to be merged 5852380-zd-woothemes

**Comment by Gustavo-Hilario** [comment]
Another report here: 5853510-zen. Unable to use the font **Open Sans** on the **2023** theme.

**Comment by daria2303** [comment]
5853470-zd-woothemes 
Munchies theme
Applied workaround

**Comment by pak-cia** [comment]
5851776-zd-woothemes
seems to be another with the same issue
suggesting workaround

**Comment by SuzzieG** [comment]
5850931 - zen

**Comment by Copons** [comment]
Heads up: the [fix for this](https://github.com/WordPress/gutenberg/pull/46944) has been merged in Core Gutenberg and will be included in Gutenberg 15.0, scheduled for next week (plus a couple more days for the Dotcom merge).

We'll try to get a point release to get the fix on Dotcom ASAP, but I can't be sure it will happen.

**Comment by dragstor** [comment]
39102668-hc

**Comment by jp-imagines** [comment]
Another report in 5849432-zd-woothemes (Twenty Twenty-Two (Blue) theme). Setting their Navigation block in different template headers to specifically use the desired font (`DM Sans` in this case) fixed things, but glad to hear a proper fix is coming soon!

**Comment by Gustavo-Hilario** [comment]
Another report in 39551784-hc **(Archeo theme)**. Please follow up with the user when this is fixed. They are using a third-party plugin for now.

**Comment by louinel** [comment]
Another report #5840854-zen-woothemes
Vivre theme.

**Comment by sophiegyo** [comment]
Another report: https://wordpress.com/forums/topic/customising-typography-is-not-reflecting-in-website/?view=all#post-3907482
Livro theme

**Comment by ccwalburn** [comment]
Another report: 5849680-zen
Archeo theme

**Comment by laragee6** [comment]
5842661-zen 
Twenty Twenty-Two

**Comment by Robertght** [comment]
Another in 5860983-zen Skatepark

**Comment by zdenys** [comment]
Another case in `5856947-zen` Twenty Twenty-Two theme

**Comment by shameemreza** [comment]
Another in 5840854-zen Twenty Twenty-Three theme

**Comment by aetta** [comment]
possibly +1 here: 5842760-zen using the blank canvas theme!

**Comment by carinapilar** [comment]
Another possible one: 39530352-hc

**Comment by carinapilar** [comment]
One more on 39268241-hc

**Comment by adelineyaw** [comment]
User report 38475234-hc (Pixl theme)

**Comment by hanananah** [comment]
Another possible one: 39525787-hc (Tsubaki theme)

**Comment by SuzzieG** [comment]
Another possible one: #5867468- zen Jackson Theme

**Comment by Copons** [comment]
Reopening this as not actually completely fixed yet. 😄 

We addressed an incompatibility issue with external font providers on JP (and Dotcom via JP) affecting live sites, but we're still waiting for Gutenberg 15.0 to land on Dotcom (it's currently available on edge sites).

So as a recap, to fix all the things on Dotcom we need:

- Gutenberg 15.0, landing on Dotcom any day now.
- The [JP fix](https://github.com/Automattic/jetpack/pull/28411) — we are asking JP folks to expedite a release on Dotcom.

**Comment by thekingsprojects** [comment]
Another case here: 5874375-zen

The workaround of setting the font on a specific block seems to have worked.

**Comment by Copons** [comment]
Gutenberg 15.0 is on Dotcom production, and I cannot reproduce this issue anymore! ✨

**Comment by vajrasar** [comment]
Hi everyone 👋 Just a quick note to let you all know @Automattic/dotcom-triage-pa-group has finished following up with affected users reported in this thread.

Thanks for the reports, please let us know here if you see any future recurrence of the issue.

**Comment by connorhipps** [comment]
@vajrasar I'm still able to replicate this issue on the following ticket: 5842760-zd-woothemes

Is there any chance the issue hasn't been fully solved yet? The site in question is a simple site on the Personal Plan.

**Comment by Copons** [comment]
@connorhipps it seems this is an issue with Blockbase, which powers a few of our themes (such as Blank Canvas Blocks, used by the site from the latest ticket).

https://github.com/Automattic/themes/issues/6832

We have routed the issue to the Themes team, as it's not about Global Styles. 🙂

**Comment by katiebethbrown** [comment]
@Copons - I'm reopening as I'm experiencing this issue on a Team51 site. It is a simple site on a Premium plan using the [Yuga theme](https://wordpress.com/theme/yuga). Yuga is not a Blockbase child theme. (pNEWy-fcl-p2#comment-54800)

Heading are set to use Libre Baskerville via Global Styles:

<img width="600px" src="https://user-images.githubusercontent.com/42279181/220758564-a9e18159-d216-4d59-95a5-aaf3659f5ae0.png">

But are rending in DM Sans, which appears to be the theme's default.

<img width="300px" src="https://user-images.githubusercontent.com/42279181/220758875-c8e0a894-5eed-4384-8d83-f3ba21918d4e.png">

I believe this is the same as the original issue, because I can see that the site title (visible in the screenshot above) _does_ use Libre Baskerville.

## Details
Gutenberg: v15.1.1
Editing Toolkit: 3.57143
WP AMP: 2.0.5

**Comment by daria2303** [comment]
5978568-zd-woothemes
theme Lettre
issue with Site title when linked and navigation, doesn't work both in Global Styles and individual block settings

**Comment by BogdanUngureanu** [comment]
Hi @katiebethbrown and @daria2303. T-Rex had a look at those issues and it seems that both issues are unrelated to this issue.

In the case of Yugo, we've noticed that it's caused because the theme styles the `core/heading` element with a font. Because of this Global Styles (font family in this case) defined in Typography > Headings are overridden. In order to style these elements from Global Styles, you'll have to go to Global Styles > **Blocks** > Heading > Typography; 

@daria2303 As for the Lettre, the font is not changed because the site title is a custom block that is also styled with a fontFamily in the theme.json definition (just like Yugo). In this case, you'll have to go to Global Styles > Blocks > Site title > Typography. 

I don't have enough context of why both themes have the fontFamily set for the blocks; maybe it's something that could be discussed with the themes team.

Since both issues are unrelated to the original issue and are theme specific, I'll be closing this issue again. :)

**Comment by daria2303** [comment]
@BogdanUngureanu thanks, nevertheless in my case this method doesn't work, as I mentioned. I have to apply CSS.

**Comment by BogdanUngureanu** [comment]
Hi @daria2303, that's strange. Can you post a video (or the steps)? I've tried it again with Lettre and I was able to configure the font for the site title:

https://user-images.githubusercontent.com/995955/220966043-6428131e-25b0-4b2e-96a9-9e714a71a5d2.mov
